Makefile

```diff
@@ -1,32 +1,58 @@
-all:
+# Default build type
+build_type?="RelWithAsserts"
+
+# Default install prefix
+prefix?="/usr/local"
+
+.PHONY: clean cleanup cmake
+
+all: igrec
+
+cmake:
+	mkdir -p build/release
+	cd build/release && cmake ../.. -DCMAKE_BUILD_TYPE="${build_type}" -DCMAKE_INSTALL_PREFIX=${prefix} -Wno-dev
+
+igrec: cmake
 	$(MAKE) -C build/release all
 
-rig:
-	$(MAKE) -C build/release/ig_repertoire_constructor ig_repertoire_constructor
+pack: igrec
+	cd build/release && cpack
 
-dig:
-	$(MAKE) -C build/debug/ig_repertoire_constructor ig_repertoire_constructor
+deb: igrec
```

Comment:
For mac I suggest
1) Homebrew formula: https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md
2) .pkg
3) .dmg

for 2 and 3 see https://guide.macports.org/chunked/using.binaries.html